app/features/welcome/components/Welcome.js

```diff
@@ -0,0 +1,145 @@
+// @flow
+
+import React, { Component } from 'react';
+import URL from 'url';
+
+// $FlowFixMe
+import styled from 'styled-components';
+
+const CenterContent = styled.div`
+    display: flex;
+    align-items: center;
+    justify-content: center;
+`;
+
+const WelcomeWrapper = CenterContent.extend`
+    background: linear-gradient(#165ecc,#44A5FF);
+    height: 100vh;
+    width: 100%;
+`;
+
+// $FlowFixMe
+import { FieldTextStateless } from '@atlaskit/field-text';
+
+// $FlowFixMe
+import Button from '@atlaskit/button';
+
+// $FlowFixMe
+import { AtlasKitThemeProvider } from '@atlaskit/theme';
+
+
+type Props = {
+
+    /**
+     * React Router history object (for flow)
+     */
+    history: Object;
+};
+
+type State = {
+
+    /**
+     * URL of the room to join
+     * If this is not a url it will be treated as room name for default domain
+     */
+    url: string;
+};
+
+/**
+ * Welcome Component.
+ */
+export default class Welcome extends Component<Props, State> {
+
+    /**
+     * Initialize Welcome Component
+     */
+    constructor(props: Props) {
+        super(props);
+
+        this.state = {
+            url: ''
+        };
+
+        // Bind event handlers
+
+        // $FlowFixMe
+        this._onURLChange = this._onURLChange.bind(this);
+
+        // $FlowFixMe
+        this._onFormSubmit = this._onFormSubmit.bind(this);
+
+        // $FlowFixMe
+        this._onJoin = this._onJoin.bind(this);
+    }
+
+    /**
+     * Render function of component.
+     *
+     * @return {ReactElement}
+     */
+    render() {
+        return (
+            <AtlasKitThemeProvider mode='light'>
+                <WelcomeWrapper>
+                    <CenterContent>
+                        <form onSubmit = { this._onFormSubmit }>
+                            <FieldTextStateless
+                                autoFocus = {true}
+                                isLabelHidden = {true}
+                                shouldFitContainer = { true }
+                                onChange = { this._onURLChange }
+                                type = 'text'
+                                value = { this.state.url } />
+                        </form>
+                        <Button
```

Comment:
This button should be disabled if the URL is not valid.

Reply:
Yeah... 
I have modified welcome page more and trying to do something like in the mobile app. using [onboarding](https://atlaskit.atlassian.com/packages/core/onboarding) component (I'll push that too when I'm done)

But there is some bug in that. I couldn't figure out if I am doing something wrong or component is behaving wrongly. 

I'll update you with this.

Reply:
Let's leave onboarding for later. For now, if the URL is not correct, we should not do anything if the user taps enter or the GO button.